### PR TITLE
Fix all non-zero delays in transform stream tests

### PR DESCRIPTION
**Pull request by ricea**
Some tests for TransformStream used the delay() function with non-zero
delays. Mostly this was done to make things happen asynchronously in a
particular order, and has been replaced with equivalent code using
promise chains.

The test "TransformStream flush is called after all queued writes
finish, once the writable is closed" was passing accidentally as it was
using test() where it should have been using promise_test(). In addition
to removing non-zero delays from the test, also fix it to use
promise_test() and remove the backpressure that was causing transform()
to never be called.

See https://github.com/whatwg/streams/issues/548 for background on this
clean-up.

### CODE REVIEW

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/20478)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/300190046)

File: streams/transform-streams/flush.js [hunk 1]
```diff
@@ -20,17 +20,20 @@ promise_test(() => {
 }, 'TransformStream flush is called immediately when the writable is closed, if no writes are queued');
 
 // TODO
```
Comment (domenic) [inline streams/transform-streams/flush.js]: Any idea what this TODO comment is doing here?

**Review by domenic — APPROVED** [review]
Really nice fixes. Approving with an opportunity for removing a stray TODO that I'll leave to you.

File: streams/transform-streams/flush.js (hunk 1 above)
Comment (ricea) [reply streams/transform-streams/flush.js]: I was trying to ignore it.

It turns out it was added by @marvinhagemeister a year ago when he ported the tests to wpt format in https://github.com/whatwg/streams/pull/531. I can find nothing in that test or the surrounding ones that needs doing, so I have removed it.

Marvin, if you remember what needed doing here, please speak up!

Comment (marvinhagemeister) [reply streams/transform-streams/flush.js]: Just went through my original PRI honestly can't remember why I put it there :/

**Pull request merged**